### PR TITLE
Add button to clear all waypoints

### PR DESCRIPTION
**Pull request by byBackfish**
Implements [Auto delete waypoints button](https://feedback.wynntils.com/posts/51/auto-delete-waypoints-button)

image

### CODE REVIEW

**Comment by ryanzhoudev** [comment]
From what I can tell, this doesn't actually delete the waypoints from cloud and they just reappear after a relaunch. I think you will need to deal with `MapConfig.Waypoints.INSTANCE.waypoints` (L288; same file).

**Comment by byBackfish** [comment]
> From what I can tell, this doesn't actually delete the waypoints from cloud and they just reappear after a relaunch. I think you will need to deal with `MapConfig.Waypoints.INSTANCE.waypoints` (L288; same file).

I thought `onWaypointChange()` would sync them? I tested it and even after restarting, they did not re-appear

**Comment by byBackfish** [comment]
I'll look into it tho. Thanks for letting me know

File: src/main/java/com/wynntils/modules/map/overlays/ui/WaypointOverviewUI.java [hunk 1]
```diff
@@ -256,6 +260,17 @@ protected void actionPerformed(GuiButton b) {
                 "Import  ==  SUCCESS",
                 String.format("Imported %d waypoints", newWaypoints)
             );
+        } else if (b == clearBtn) {
+            clearCount++;
+            if(clearCount == clearCountMax){
```
Comment (ryanzhoudev) [inline src/main/java/com/wynntils/modules/map/overlays/ui/WaypointOverviewUI.java]:
```suggestion
            if (clearCount == clearCountMax){
```

File: src/main/java/com/wynntils/modules/map/overlays/ui/WaypointOverviewUI.java [hunk 2]
```diff
@@ -50,6 +51,8 @@ public class WaypointOverviewUI extends GuiScreen {
     private int pageHeight;
     private int group = ungroupedIndex;
     private int groupWidth;
+    int clearCount = 0;
+    final int clearCountMax = 3;
```
Comment (ryanzhoudev) [inline src/main/java/com/wynntils/modules/map/overlays/ui/WaypointOverviewUI.java]: Explicitly declare private for consistency

File: src/main/java/com/wynntils/modules/map/overlays/ui/WaypointOverviewUI.java [hunk 3]
```diff
@@ -76,6 +79,7 @@ public void initGui() {
 
         this.buttonList.add(exportBtn = new GuiButton(8, this.width/2 + 26, this.height - 45, 50, 20, "EXPORT"));
         this.buttonList.add(importBtn = new GuiButton(9, this.width/2 - 76, this.height - 45, 50, 20, "IMPORT"));
+        this.buttonList.add(clearBtn = new GuiButton(9, 25, this.height - 45, fontRenderer.getStringWidth("CLEAR (Press " + clearCountMax + " times)") + 15, 20, "CLEAR (Press " + clearCountMax + " times)"));
```
Comment (ryanzhoudev) [inline src/main/java/com/wynntils/modules/map/overlays/ui/WaypointOverviewUI.java]: Explicit width declaration or wrap the two usages of that string into one variable

**Review by ryanzhoudev — COMMENTED** [review]
Last few things, then I think it's good

**Review by ryanzhoudev — DISMISSED** [review]

File: src/main/java/com/wynntils/modules/map/overlays/ui/WaypointOverviewUI.java [hunk 4]
```diff
@@ -256,6 +261,16 @@ protected void actionPerformed(GuiButton b) {
                 "Import  ==  SUCCESS",
                 String.format("Imported %d waypoints", newWaypoints)
             );
+        } else if (b == clearBtn) {
+            clearCount++;
+            if(clearCount == clearCountMax){
```
Comment (kristofbolyai) [inline src/main/java/com/wynntils/modules/map/overlays/ui/WaypointOverviewUI.java]: Space after if.

Comment (kristofbolyai) [reply src/main/java/com/wynntils/modules/map/overlays/ui/WaypointOverviewUI.java]: And before {

File: src/main/java/com/wynntils/modules/map/overlays/ui/WaypointOverviewUI.java [hunk 5]
```diff
@@ -256,6 +261,16 @@ protected void actionPerformed(GuiButton b) {
...
+                waypoints.clear();
+                onWaypointChange();
+                clearCount = 0;
+                clearBtn.displayString = "CLEARED!";
+            } else {
+                clearBtn.displayString = "CLEAR (Press " + (clearCountMax-clearCount) + " times)";
```
Comment (kristofbolyai) [inline src/main/java/com/wynntils/modules/map/overlays/ui/WaypointOverviewUI.java]: clearCountMax - clearCount

**Review by kristofbolyai — CHANGES_REQUESTED** [review]

**Review by ryanzhoudev — DISMISSED** [review]

**Comment by ryanzhoudev** [comment]
You should fix merge conflicts before requesting review since the review will be dismissed after the resolve

**Review by kristofbolyai — APPROVED** [review]

**Review by ryanzhoudev — APPROVED** [review]

**Pull request merged**